src/Components/ClustersListTable/ClustersListTable.spec.ct.js

```diff
@@ -59,51 +66,120 @@ let namedClustersDefaultSorting = _.orderBy(
+
+// TODO add more combinations of filters for testing
+const filterCombos = [{ risk: ['Critical', 'Moderate'], name: 'foo' }];
+
+// TODO use filterData in all tests except of data or namedClusters or namedClustersDefaultSorting
+
 describe('data', () => {
   it('has values', () => {
-    cy.wrap(data).its('length').should('be.gte', 1);
+    cy.wrap(filterData(data)).its('length').should('be.gte', 1);
   });
   it('has more entried than default pagination', () => {
-    cy.wrap(data).its('length').should('be.gt', DEFAULT_ROW_COUNT);
+    cy.wrap(filterData(data)).its('length').should('be.gt', DEFAULT_ROW_COUNT);
   });
   it('at least one cluster has cluster name', () => {
-    cy.wrap(_.filter(data, (it) => it.cluster_name))
+    cy.wrap(_.filter(filterData(data), (it) => it.cluster_name))
       .its('length')
       .should('be.gte', 1);
   });
   it('first cluster has name', () => {
-    cy.wrap(data[0]['cluster_name']).should('not.be.empty');
+    cy.wrap(filterData(data)[0]['cluster_name']).should('not.be.empty');
   });
   it('first page items contains at least one cluster without name', () => {
     const itemsInFirstPage = DEFAULT_DISPLAYED_SIZE;
-    cy.wrap(_.filter(data.slice(0, itemsInFirstPage), (it) => it.cluster_name))
+    cy.wrap(
+      _.filter(
+        filterData(data).slice(0, itemsInFirstPage),
+        (it) => it.cluster_name
+      )
+    )
       .its('length')
       .should('be.lt', itemsInFirstPage);
   });
   it('at least one entry has last seen', () => {
-    cy.wrap(_.filter(data, (it) => it.last_checked_at))
+    cy.wrap(_.filter(filterData(data), (it) => it.last_checked_at))
       .its('length')
       .should('be.gte', 1);
   });
   it('at least one entry does not have last seen', () => {
-    cy.wrap(_.filter(data, (it) => it.last_checked_at === undefined))
+    cy.wrap(
+      _.filter(filterData(data), (it) => it.last_checked_at === undefined)
+    )
       .its('length')
       .should('be.gte', 1);
   });
   it('at least one entry does not have all values for total risk categories', () => {
     cy.wrap(
-      _.filter(data, (it) => Object.keys(it['hits_by_total_risk']).length)
+      _.filter(
+        filterData(data),
+        (it) => Object.keys(it['hits_by_total_risk']).length < 4
+      )
     )
       .its('length')
       .should('be.gte', 1);
   });
+  it('at least two clusters match foo for their names', () => {
+    cy.wrap(filterData(namedClusters, { name: 'foo' }))
+      .its('length')
+      .should('be.gt', 1);
+  });
+  it('only one cluster matches foo bar in the name', () => {
+    cy.wrap(filterData(namedClusters, { name: 'foo bar' }))
+      .its('length')
+      .should('be.eq', 1);
+  });
+  it('the first combo filter has less clusters hitting that the default and at least one', () => {
+    cy.wrap(filterData(namedClusters, filterCombos[0]))
+      .its('length')
+      .should('be.gte', 1)
+      .and('be.lt', filterData(namedClusters, {}).length);
```

Comment:
You are right. On the other hand, my plan is to have namedClusters as default parameter for filterData instead of having to pass it all the time. But I need to re-check all the data usage in the file.
If you are fine with it, I will put a TODO